### ISSUE TITLE
several logging issues

### ISSUE THREAD

**Issue by euri10** [Kludex/uvicorn#644]
1. first issue is minor and undetected currently because the default formatter don't output the logger's name, but on  h11 implementation the uvicorn.access logger output comes from the uvicorn.error logger.

so if your formatter has a `%(name)s` in it you'll see uvicorn.error instead of uvicorn.access in all your access logs, 

see below:

https://github.com/encode/uvicorn/blob/a839da2564e5168c6e99c3b22819950f26f7d178/uvicorn/protocols/http/h11_impl.py#L362

2. the colorized formatter operates and modifies the logging records (https://github.com/encode/uvicorn/blob/e55bd89184bb3f47d7024c3dd6538ecd68ebe114/uvicorn/logging.py#L45-L54), which has non-desirable side-effect since other handlers may use it down the line

for instance I can output this:

```
TRACE:    root - 33 - [SFJAPI]- ASGI [1] Started scope={'type': 'lifespan'}
TRACE:    root - 33 - [SFJAPI]- ASGI [1] Receive {'type': 'lifespan.startup'}
INFO:     core.events - 16 - [SFJAPI]- Creating pool
DEBUG:    core.events - 17 - [SFJAPI]- postgresql://postgres:postgres@localhost:5439/postgres
INFO:     core.events - 19 - [SFJAPI]- Pool created
```
by just adding 

```
  app_format_colored:
    "()": "core.logging.AppColourizedFormatter"
    format: "%(levelprefix)s %(name)s - %(lineno)d - [SFJAPI]- %(message)s"
    use_colors: True
```

it shows clearly that the default uvicorn.error logger gets polluted by my formatter since it receives the `SFJAPI` string that is set on my custom format.

3. the `uvicorn.asgi` logger wrongly uses the root logger as shown in [here](https://github.com/encode/uvicorn/blob/608866d3b6dea49f4f8ff28ed19398b0ae47faad/uvicorn/middleware/message_logger.py#L33) `logging.log(TRACE_LOG_LEVEL, message, *args, **kwargs)` , you can see it in the log output above, `root 33` is the logger's name and the line in python standard library logging, this is a simple fix and root logger shouldn't be touched.

PR incoming

**Comment by euri10** [comment]
without the fix with the logging config at the bottom it output this:

<details>
  <summary>problematic output</summary>

```
TRACE:    root - 33 - [SFJAPI]- ASGI [1] Started scope={'type': 'lifespan'}
TRACE:    root - 33 - [SFJAPI]- ASGI [1] Receive {'type': 'lifespan.startup'}
INFO:     core.events - 16 - [SFJAPI]- Creating pool
DEBUG:    core.events - 17 - [SFJAPI]- postgresql://postgres:postgres@localhost:5439/postgres
INFO:     core.events - 19 - [SFJAPI]- Pool created
INFO:     Started server process [27984]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
TRACE:    root - 33 - [SFJAPI]- ASGI [1] Send {'type': 'lifespan.startup.complete'}
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
TRACE:    root - 33 - [SFJAPI]- 127.0.0.1:42742 - ASGI [2] Started scope={'type': 'http', 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'client': ('127.0.0.1', 42742), 'scheme': 'http', 'method': 'POST', 'root_path': '', 'path': '/auth/', 'raw_path': b'/auth/', 'query_string': b'', 'headers': '<...>'}
INFO:     simplefapi - 43 - [SFJAPI]- test
TRACE:    root - 33 - [SFJAPI]- 127.0.0.1:42742 - ASGI [2] Receive {'type': 'http.request', 'body': '<0 bytes>', 'more_body': False}
TRACE:    root - 33 - [SFJAPI]- 127.0.0.1:42742 - ASGI [2] Send {'type': 'http.response.start', 'status': 200, 'headers': '<...>'}
INFO:     127.0.0.1:42742 - 'POST /auth/ HTTP/1.1' 200 OK
TRACE:    root - 33 - [SFJAPI]- 127.0.0.1:42742 - ASGI [2] Send {'type': 'http.response.body', 'body': '<9 bytes>'}
TRACE:    root - 33 - [SFJAPI]- 127.0.0.1:42742 - ASGI [2] Completed
TRACE:    127.0.0.1:42742 - Connection made
INFO:     Shutting down
TRACE:    127.0.0.1:42742 - Connection lost
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [27984]
TRACE:    root - 33 - [SFJAPI]- ASGI [1] Receive {'type': 'lifespan.shutdown'}
INFO:     core.events - 26 - [SFJAPI]- Closing pool
INFO:     core.events - 28 - [SFJAPI]- Pool closed
TRACE:    root - 33 - [SFJAPI]- ASGI [1] Send {'type': 'lifespan.shutdown.complete'}
TRACE:    root - 33 - [SFJAPI]- ASGI [1] Completed
```
</details>

with the PR it ouputs everything correctly it seems:

<details>
  <summary>fixed output</summary>

```
TRACE:    ASGI [1] Started scope={'type': 'lifespan'}
TRACE:    ASGI [1] Receive {'type': 'lifespan.startup'}
INFO:     core.events - 16 - [SFJAPI]- Creating pool
DEBUG:    core.events - 17 - [SFJAPI]- postgresql://postgres:postgres@localhost:5439/postgres
INFO:     Started server process [32906]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     core.events - 19 - [SFJAPI]- Pool created
TRACE:    ASGI [1] Send {'type': 'lifespan.startup.complete'}
TRACE:    127.0.0.1:43138 - Connection made
TRACE:    127.0.0.1:43138 - ASGI [2] Started scope={'type': 'http', 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'client': ('127.0.0.1', 43138), 'scheme': 'http', 'method': 'POST', 'root_path': '', 'path': '/auth/', 'raw_path': b'/auth/', 'query_string': b'', 'headers': '<...>'}
INFO:     simplefapi - 43 - [SFJAPI]- test
TRACE:    127.0.0.1:43138 - ASGI [2] Receive {'type': 'http.request', 'body': '<0 bytes>', 'more_body': False}
TRACE:    127.0.0.1:43138 - ASGI [2] Send {'type': 'http.response.start', 'status': 200, 'headers': '<...>'}
INFO:     127.0.0.1:43138 - 'POST /auth/ HTTP/1.1' 200 OK
TRACE:    127.0.0.1:43138 - ASGI [2] Send {'type': 'http.response.body', 'body': '<9 bytes>'}
TRACE:    127.0.0.1:43138 - ASGI [2] Completed
```
</details>

<details>
  <summary>logging config</summary>
    
  ```yaml
version: 1
disable_existing_loggers: False
formatters:
  uvicorn_default:
    "()": "uvicorn.logging.DefaultFormatter"
    fmt: "%(levelprefix)s %(message)s"
    use_colors: True
  uvicorn_access:
    "()": "uvicorn.logging.AccessFormatter"
    fmt: "%(levelprefix)s %(client_addr)s - '%(request_line)s' %(status_code)s"
    use_colors: True
  app_format_colored:
    "()": "core.logging.AppColourizedFormatter"
    format: "%(levelprefix)s %(name)s - %(lineno)d - [SFJAPI]- %(message)s"
    use_colors: True
handlers:
  uvicorn_default:
    formatter: uvicorn_default
    class: "logging.StreamHandler"
    stream: "ext://sys.stderr"
  uvicorn_access:
    formatter: uvicorn_access
    class: "logging.StreamHandler"
    stream: "ext://sys.stdout"
  uvicorn_asgi:
    formatter: uvicorn_default
    class: "logging.StreamHandler"
    stream: "ext://sys.stdout"
  console_app:
    class: logging.StreamHandler
    formatter: app_format_colored
    stream: ext://sys.stdout
loggers:
  uvicorn.error:
    level: DEBUG
    handlers: [uvicorn_default]
    propagate: no
  uvicorn.access:
    level: DEBUG
    handlers: [uvicorn_access]
    propagate: no
  uvicorn.asgi:
    level: TRACE
    handlers: [uvicorn_asgi]
    propagate: no
  simplefapi:
    level: DEBUG
    handlers: [console_app]
    propagate: no
root:
  level: INFO
  handlers: [console_app]
  ```
  
</details>

**Comment by euri10** [comment]
fixed by #645
